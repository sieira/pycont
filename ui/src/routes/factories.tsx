import React from 'react'
import { Route, Redirect } from 'react-router-dom'

import { AuthStateRouteProps } from './types'

export const authStateRouteFactory = (
  authenticatedValue: boolean,
  defaultFallbackRoute: string,
  routeHeader: JSX.Element,
  routeFooter: JSX.Element
): React.FC<AuthStateRouteProps> => ({
  component: Component,
  isAuthenticated,
  ...otherProps
}: AuthStateRouteProps): JSX.Element => {
  if (isAuthenticated !== authenticatedValue) {
    return <Redirect push to={defaultFallbackRoute} />
  }

  return (
    <>
      {routeHeader}
      <Route
        render={(otherProps): JSX.Element => (
          <>
            <Component {...otherProps} />
          </>
        )}
      />
      {routeFooter}
    </>
  )
}
