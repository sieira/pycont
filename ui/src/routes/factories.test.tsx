import React from 'react';

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { authStateRouteFactory } from './factories';

configure({adapter: new Adapter()});

it('Should add header and footer', () => {
  const header = <div>Header</div>;
  const footer = <div>Footer</div>
  const route = authStateRouteFactory(true, '/fallback', header, footer);

  const component = <div id="i-am-muzzy"/>
  const componentRoute = route(component, true);

  expect(componentRoute).toMatchSnapshot();
  expect(componentRoute.props.children[0]).toStrictEqual(header);
  expect(componentRoute.props.children[2]).toStrictEqual(footer);
});
