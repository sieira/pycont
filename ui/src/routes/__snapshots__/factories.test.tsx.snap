// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should add header and footer 1`] = `
<React.Fragment>
  <div>
    Header
  </div>
  <Route
    render={[Function]}
  />
  <div>
    Footer
  </div>
</React.Fragment>
`;
